packages/data/src/lib/data-view/data-view.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { CartoError } from '@carto/toolkit-core';
+import { Layer } from '@carto/toolkit-viz';
+import { DataView } from './data-view';
+import { AggregationTypes } from '../aggregations/aggregations';
+
+describe('DataView', () => {
+  describe('Instance Creation', () => {
+    it('should create new DataView instance', () => {
+      expect(
+        () => new DataView(new Layer('fake_source'), 'fake_column')
+      ).not.toThrow();
+    });
+
+    it('should throw an exception when source is not provided', () => {
+      expect(() => new DataView(undefined as never, 'fake_column')).toThrow(
+        new CartoError({
+          type: '[DataView]',
+          message: 'Source was not provided while creating dataview'
+        })
+      );
+    });
+
+    it('should throw an exception when column is not provided', () => {
+      expect(
+        () => new DataView(new Layer('fake_source'), undefined as never)
+      ).toThrow(
+        new CartoError({
+          type: '[DataView]',
+          message: 'Column name was not provided while creating dataview'
+        })
+      );
+    });
+  });
+
+  describe('Events', () => {
+    it('dataUpdate', () => {
+      const layer = new Layer('fake_source');
+      const dataView = new DataView(layer, 'popEst');
+
+      const dataUpdateSpy = jest.fn();
+      dataView.on('dataUpdate', dataUpdateSpy);
+
+      layer.emit('viewportLoad');
+
+      expect(dataUpdateSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('aggregate', () => {
+    it('should return an empty object if operations are not provided', () => {
+      const dataView = new DataView(new Layer('fake_source'), 'fake_column');
+      expect(dataView.aggregate()).toMatchObject({});
```

Comment:
I'm not sure this use case makes sense. If we have no operations, what's the purpouse of the dataview?